src/algorithm.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+// Pragma once is supported on all modern compilers
+// TODO: decide whether to use include guards anyway
+
+#include <algorithm>
+
+//--------------------------------------------------------------------------------------------------
+// Container-wide versions of std algoithms.
+//--------------------------------------------------------------------------------------------------
+
+template <typename Container, typename Predicate>
```

Comment:
I'm not sure such thin wrappers are worthwhile, as you know we already have pretty bad build times, and I'm not really a fan of wrappers that merely hide what's happening.